rust/src/measurements/tulap/make_tulap.tex

```diff
@@ -0,0 +1,154 @@
+\begin{defn}\label{def2}  % Def 3.1
+    \cite[Definition~3.1]{awan2023canonical} \\
+    Let $f$ be a symmetric nontrivial tradeoff function. A {continuous} distribution function $F$ is a \emph{canonical noise distribution} (CND) for $f$ if 
+    \begin{enumerate}[(1)]
+        \item for every statistic $S: X^n\rightarrow \mathbb{R}$ with sensitivity $\Delta>0$, and $N\sim F(\cdot)$, 
+            the mechanism $S(X) + \Delta N$ satisfies $f$-DP. Equivalently, for every $m\in [0,1]$, $T(F(\cdot),F(\cdot-m))\geq f$,
+        \item $f(\alpha)=T(F(\cdot),F(\cdot-1))(\alpha)$ for all $\alpha \in (0,1)$,
+        \item $T(F(\cdot),F(\cdot-1))(\alpha) = F(F^{-1}(1-\alpha)-1)$ for all $\alpha \in (0,1)$,
+        \item $F(x) = 1-F(-x)$ for all $x\in \mathbb{R}$; that is, $F$ is the cdf of a random variable which is symmetric about zero.
+    \end{enumerate}
+\end{defn}
+
+\begin{defn} \label{def1}  % Def 3.7
+    \cite[Definition~3.7]{awan2023canonical} \\
+    Let $f$ be a symmetric nontrivial tradeoff function, and let {$c\in [0,1/2)$} be the unique fixed point of $f$: $f(c)=c$. 
+    We define $F_f:\mathbb{R}\rightarrow \mathbb{R}$ as  \[ F_f(x) = \begin{cases}
+        f(1-F_f(x+1))&x<-1/2\\
+        c(1/2-x) + (1-c)(x+1/2)&-1/2\leq x\leq 1/2\\
+        1-f(F_f(x-1))&x>1/2.\\
+    \end{cases}\]
+\end{defn}
+In Definition~\ref{def1}~\parencite{awan2023canonical}, 
+the fact that there is a unique fixed point follows from the fact that $f$ is convex and decreasing $f$-DP~\parencite{dong2019gaussian}, 
+and so intersects the line $y=\alpha$ at a unique value. 
+Note that in Definition~\ref{def1}~\parencite{awan2023canonical}, 
+the cumulative distribution function (CDF) corresponds to a uniform random variable on the interval $[-1/2,1/2]$. 
+However, due to the recursive nature of $F_f$ and the fact that $f$ is generally non-linear, 
+the Canonical Noise Distribution (CND) from Definition~\ref{def1}~\parencite{awan2023canonical} need not be uniformly distributed on any other intervals.
+
+Theorem~\ref{thm1}~\parencite{awan2023canonical} below states that for any nontrivial tradeoff function, 
+this construction yields a CND, which can be constructed as in Definition~\ref{def1}~\parencite{awan2023canonical}. 
+This CND can be used to add perfectly calibrated noise to a statistic to achieve $f$-DP~\parencite[Proposition~2.2]{dong2019gaussian}. 
+
+\begin{thm} \label{thm1}  % Thm 3.9
+    \cite[Theorem~3.9]{awan2023canonical} \\
+    Let $f$ be a symmetric nontrivial tradeoff function and let $F_f$ be as in Definition~\ref{def1}~\parencite{awan2023canonical}. 
+    Then, as stated in \parencite[Theorem~3.9]{awan2023canonical}, $F_f$ is a canonical noise distribution for $f$. 
+\end{thm}
+
+In Definition~\ref{def1}~\parencite{awan2023canonical}, we explained the cdf of the CND we made. 
```

Comment:
Definition~\ref{def1} explains...

No need to cite the paper over and over again if you've already done that in the definition above.